### DISCUSSION TITLE
Is it able to embed pyo3 and python code into a slint app?

### DISCUSSION THREAD

**Discussion by DennisZhangOiler** [slint-ui/slint#6734] [Q&A] [answered]
And build&distribute it over windows, linux and android.

**Comment by hunger** [comment]
Is that a Slint up with a business logic written in Rust or one using python? Anyway: Both should work.

Whether its fine to distribute your code over Windows, Linux and Android: I don't know, as I do not know what code you use besides Slint! Please check that yourself.

Looking at the licensing problem from the Slint side only: Of course you can distribute under the terms of the relevant license you have chosen to use Slint under.

**Reply by DennisZhangOiler** [reply]
I am just execited to find out slint and want to try it out, currently we are using web based cross platform frameworks, it has some performance issues and hard to do some work that should not be done in web, I am looking for a better replacement then I found slint, it can support windows, linux and android on the way.

**Reply by hunger** [reply]
We also support different backend languages: JS/TS and python are not final yet, but especially JS/TS with bun supposedly is in a very good state. When you have devs comfortable with the web that might be an inetersting option for you, too.

**Reply by DennisZhangOiler** [reply]
I just checkout the plotter example, it seems to be a callback with variable updated? will I be able to draw images manually at 30 FPS? it's processed images from cameras.

**Reply by hunger** [reply, marked as answer]
> I just checkout the plotter example, it seems to be a callback with variable updated? will I be able to draw images manually at 30 FPS? it's processed images from cameras.

That most depends on how fast you can generate the images.

You might find the `examples/gstreamer-player` example interesting as well for this use-case.
